### ISSUE TITLE
Quarantine for identification name terms (misspelling, no longer accepted, etc.)

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#2927]
>misspelled in the article you linked to

Yes, exactly.

* If someone coming from that publication searches Arctos and we've deleted this, they're just at a dead end
* If someone coming from that publication searches Arctos and we've made a relationship to the correct spelling, they'll get where they need to be.

That's not a very good example - doesn't seem like the kind of publication that would lead people to Arctos - but I still think it's worth fleshing out as a matter of policy. Is getting users where they need to be a "community service" we're willing to provide, or should we delete all taxonomy for which we don't have an immediate and compelling use for, or ?????????????

>I vote to delete the misspelling so that we won't be adding to the problem.

I could be convinced to go that way, but https://github.com/ArctosDB/arctos/issues/2499 provides a potential mechanism to both preserve the name and to isolate it so we're not accidentally perpetuating funky spellings.

_Originally posted by @dustymc in https://github.com/ArctosDB/arctos/issues/2925#issuecomment-658987745_

**Comment by Jegelewicz** [comment]
Suggest that rather than delete misspelled taxa, they be merged to the correct name with notifications just as we do with agents.

Upon merger, all records with IDs using the "bad duplicate" name will be updated to the "good" name with the "bad" name placed in identification remark. (I advocate for making the first ID A {string}, then creating a second ID...)

BUT

1. Person suggesting the merge has to supply a reason for the merge.
2. If the merge completes, the "bad duplicate" taxon name is saved along with the reason given above and a relationship to the "good" name that can be used in searches, but NOT in data entry.
3. Data entry of the "bad" name will result in a suggestion to use the "good" one along with the reason given for the "bad" designation in the first place.

Reasoning for this treatment:

> Another issue is that spelling errors from the literature may be entered into databases, perhaps the error may occur during data entry, and then perpetuated, sometimes unknowingly, but on other occasions intentionally. For example the spelling error Ammothea sextarticulata (instead of Ammothea sexarticulata Munilla, 1990) was first published in 1994, and was entered as such in WoRMS in 2005. It was later corrected by the taxonomic editor but has already had 410 Google hits, whereas the correct spelling has had only 118 (checked 19 April 2012). In this case, the incorrect spelling is present in several online databases and continues to perpetuate in the literature; even the original author used the misspelling in 2008 [41]. To enable tracking of such errors, a taxonomic database should retain all published spellings but indicate which are in error.

from [Global Coordination and Standardisation in Marine Biodiversity through the World Register of Marine Species (WoRMS) and Related Databases](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3541386/)

**Comment by dustymc** [comment]
I don't get it.

>Suggest that rather than delete misspelled taxa, they be merged to the correct name with notifications just as we do with agents.

That's a delete.

> To enable tracking of such errors, a taxonomic database should retain all published spellings but indicate which are in error.

That's always been my point!

**Comment by Jegelewicz** [comment]
And we DO NOT INDICATE which are in error - especially to those doing data entry - in any meaningful way.

**Comment by dustymc** [comment]
>DO NOT INDICATE which are in error

https://github.com/ArctosDB/arctos/issues/2499 will do that in the next release.

**Comment by sharpphyl** [comment]
I'm not sure where to leave this comment - here or on #2645 

Obvious typo misspellings that are in use have to be corrected before we can delete them.  I see no reason to leave them in the database as it creates fertile ground for creating the same typo again.  Teresa's approach would, I think, be a solution to cleaning up these identifications so we can remove the misspelled name.  I've been creating annotations but I don't think they necessarily are sent to the curator or, if they are, they are ignored.  Here's the most recent one.  I'm not sure I understand the quarantine process (which was mentioned somewhere but I had never heard of) which may be an interim solution.  If we implemented #2125 it would solve the problem for collections using WoRMS (via Arctos) but not for others.

![Screen Shot 2020-07-19 at 8 46 43 AM](https://user-images.githubusercontent.com/15368365/87954637-dc9ee880-ca69-11ea-9ecd-82477ccfe441.png)

**Comment by dustymc** [comment]
https://github.com/ArctosDB/documentation-wiki/issues/156

https://github.com/ArctosDB/arctos/issues/2937

I don't have a solution for unresponsive or nonexistent Curators - that's an administrative problem. Switching to an "A {string}" identification using a similar taxa so the name can be quarantined wouldn't lose much information - perhaps we need some clause allowing that in the MOUs or etc.

**Comment by sharpphyl** [comment]
@Jegelewicz   Can you notify NMNHM of this misspelled taxon so I can quarantine it so another collection doesn't trip into this misspelling?  Thanks...and we need a better way to do this which is why I support some type of auto-update like this merger.  Unlike you, most people don't respond to an annotation or I don't know whom to notify.

@dustymc I can't quarantine a name that is actively used but can I quarantine it if it's a previous identification?  In other words, what is the definition of a "used" name?  I don't want the erroneous ID but some collections might want to retain it but don't want to leave it as an active taxon name.  

![Screen Shot 2020-07-22 at 9 22 43 AM](https://user-images.githubusercontent.com/15368365/88195973-e5282800-cbfd-11ea-8e1e-e609c939643e.png)

**Comment by dustymc** [comment]
>what is the definition of a "used" name? 

Taxon_name.taxon_name_id is used as a foreign key somewhere. Previous IDs definitely count.

>support some type of auto-update

And when it turns out that NMNHM's specimen was the type and everybody else has it wrong, but now we've trashed that information? I just can't see a safe automation of this; I think it has to come down to people working together.

**Comment by sharpphyl** [comment]
> Taxon_name.taxon_name_id is used as a foreign key somewhere. Previous IDs definitely count.

Is it possible to NOT count previous IDs for purposes of quarantining a name?  

I would like to quarantine _Conus chaldeus_.  It is a misspelling of _Conus chaldaeus_.  Only DMNS:Inv has records using it

![Screen Shot 2020-07-24 at 11 46 42 AM](https://user-images.githubusercontent.com/15368365/88420635-6d483200-cda4-11ea-8c87-c4e286cea02d.png)

We have 16 catalog records that were identified with the misspelled name, then updated with "revised taxonomy." Quarantining the name would help prevent using it again, but because we used it, I cannot quarantine it.  

![Screen Shot 2020-07-24 at 11 46 31 AM](https://user-images.githubusercontent.com/15368365/88420666-79cc8a80-cda4-11ea-8798-21dff2a93ba3.png)

I have to manually edit all the entries to delete the misspelled name first before I can quarantine it.  I've done that a few times but it's time consuming.

**Comment by dustymc** [comment]
>Is it possible to NOT count previous IDs for purposes of quarantining a name?

Ah, the ol' California Quarantine....

Short answer is no. There are lots of "soft quarantine" options that could be done via classifications, but they can't actually solve the problem. Any middle ground option provides workarounds with which a creative user could almost certainly ignore the supposed restrictions. 

I could update those to A {string} IDs pointing to the valid name or something of the sort if you want, assuming they're not overly complex IDs. I just need bad and good taxon_name_ids.

**Comment by Jegelewicz** [comment]
> I could update those to A {string} IDs pointing to the valid name or something of the sort if you want, assuming they're not overly complex IDs. I just need bad and good taxon_name_ids.

@sharpphyl I think this is the way to go.

And then perhaps we need a pathway for these things - some way for someone besides Dusty to change the old IDs to A {string}

**Comment by sharpphyl** [comment]
I'll probably do this group manually, but it would be helpful to be able to modify an existing group of identifications as well as add a new one.

**Comment by Jegelewicz** [comment]
@sharpphyl I think you could use manage identification from the search results to accomplish this in bulk. I think....we could meet up and try one together if you want.

**Comment by lin-fred** [comment]
What does quarantining a taxon do?

I switched over the list you gave me @sharpphyl but I kept the old names to check our original data. Our original data also had the misspelling (so it wasn't a typo on your end). I think the names were published at one point with the misspelling, and those publications were used in our collection as well. I think it's important to keep this information because anything to do with these specimens would have been done with that misspelling. 

So is quarantining the solution?

**Comment by dustymc** [comment]
>What does quarantining a taxon do?

Makes the name unavailable for everything except search; it's a much-preferred alternative to deletion.

>solution

Not if you're using the name in IDs - you'll have to try to convince people not to incorrectly use it via classifications.

**Comment by Jegelewicz** [comment]
@lin-fred those misspelled names used in identifications should be converted to the A {string} formula so that the name can be quarantined (if it is in use, it cannot be quarantined).

However, I have been noodling on this and wondering if we shouldn't look at "quarantine" in a different way? Instead of the Herculean effort of getting anyone using a term that should be quarantined to remove it form any identifications, perhaps quarantine should instead mean a warning when anyone is attempting to use it and a warning on any identification that is using it (a red box!)....

**Comment by dustymc** [comment]
A name can't be quarantined if it's used in identifications, and can't be used in identifications if it's quarantined. So it's not "won't show up" but "you cannot create that situation."

**Comment by dustymc** [comment]
>warning

https://github.com/ArctosDB/arctos/issues/2927#issuecomment-663670279

I still have no interest in a California Quarantine option - it still doesn't do anything that can't be done in classifications.

There's no new urgency attached to this functionality - it's a way to preserve a name that would otherwise be deleted. If you were going to clean these up and delete taxa, and those taxa could be useful in finding or understanding something, then you should quarantine them instead of deleting them. That's it.

**Comment by Jegelewicz** [comment]
But I thought the red box idea would be appealing!

**Comment by dustymc** [comment]
I do like red boxes....

**Comment by Jegelewicz** [comment]
I think we should bring to the AWG the idea that when a request is made to quarantine a name, it goes through a process (just like code table changes) and if approved, all IDs using the name will be converted to A {string}, with A being the appropriate "accepted" name.

Sound like something reasonable?

**Comment by dustymc** [comment]
>reasonable

1. I think it's often not as clear-cut as it might seem in the moment - things like https://github.com/ArctosDB/arctos/issues/3026 get debated in the literature literally for centuries, and someone declaring "you're all wrong!," however loudly, should not always be a reason for action, and
2. I'm incredibly hesitant to do anything that could conceivably change the meaning of data without direct guidance from the collections, and what looks like slight variations in a taxon name can absolutely carry a lot of information. 

I like the idea, but it probably must have a "...and when the DBAs have confirmation from all involved collections..." step.

**Comment by Jegelewicz** [comment]
Which leaves us in the same boat as we are now. I think we should at least create a name quarantine issue template. I'll draft it up.

**Comment by Jegelewicz** [comment]
Well, I do not have access to "Features" in this repository anymore. @dustymc can you create this template? Edit if necessary.

title placeholder = Taxon Name Quarantine Request - NAME

Issue Documentation is at http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html
Name Quarantine Documentation is at https://arctos.database.museum/info/ctDocumentation.cfm?table=cttaxon_name_type

The name suggested for quarantine is:

The reason for quarantine is:

The suggested treatment for all identifications currently using this name is:

Collections currently using this name in identifications include:

**Priority**
Please assign a priority-label.

**Comment by dustymc** [comment]
>do not have access to "Features"

You probably should - I think maybe @mkoo was cleaning house??

Did you just create a "tc1" branch?

**Comment by Jegelewicz** [comment]
She did clean house - I can't add people to the repository anymore either.

**Comment by dustymc** [comment]
Yep, just been dealing with a lot of malicious traffic lately (need TACC security's full and undivided attention? Just toss a few million bitcoin requests from Russian IPs in your server logs!) and panicked a bit when I saw it pop up....

>can't add people

@mkoo can you restore Teresa's superpowers?
